### DISCUSSION TITLE
Process don't close when run in ProcessTimedOutException

### DISCUSSION THREAD

**Discussion by nguyenquang1995** [spatie/browsershot#555] [Q&A] [answered]
In callBrowser() function if run in timeout case, throw ProcessTimedOutException but don't clear process

**Comment by ernst77** [comment]
Is there a nice way to kill it? Sometimes I noticed process just hangs for no reason, and all pending jobs fail because there is aleady running chormium node process which hangs... I guess I can do `exec("kill -9 $pid");`

**Comment by pharixces** [comment, marked as answer]
We had a similar issue, turned out to be unreadable cache, a bug in chromium (perhaps a race condition or something like that). Disabling browsercache or downgrading puppet (to 9.x) solved these issues for us.

**Reply by adrum** [reply]
Downgrading puppeteer to v9 worked perfectly for us, too.
